packages/stream/solana/StreamClient.ts

```diff
@@ -371,19 +371,44 @@ export default class SolanaStreamClient extends BaseStreamClient {
     const errors: ICreateMultiError[] = [];
     const signatures: string[] = [];
     const batch: BatchItem[] = [];
+    const instructionsBatch: {
+      ixs: TransactionInstruction[];
+      metadata: Keypair | undefined;
+      recipient: string;
+    }[] = [];
     metadataPubKeys = metadataPubKeys || [];
 
     for (let i = 0; i < recipients.length; i++) {
       const recipientData = recipients[i];
-      const { tx, metadataPubKey } = await this.prepareStreamTransaction(recipientData, data, {
-        sender,
-        metadataPubKeys: metadataPubKeys[i] ? [metadataPubKeys[i]] : undefined,
-      });
+      const { ixs, metadata, metadataPubKey } = await this.prepareStreamInstructions(
+        recipientData,
+        data,
+        {
+          sender,
+          metadataPubKeys: metadataPubKeys[i] ? [metadataPubKeys[i]] : undefined,
+        }
+      );
 
       metadataToRecipient[metadataPubKey.toBase58()] = recipientData;
 
       metadatas.push(metadataPubKey.toBase58());
-      batch.push({ tx, recipient: recipientData.recipient });
+      instructionsBatch.push({ ixs, metadata, recipient: recipientData.recipient });
+    }
+
+    const commitment =
+      typeof this.commitment == "string" ? this.commitment : this.commitment.commitment;
+    const hash = await this.connection.getLatestBlockhash(commitment);
```

Comment:
since this request moved earlier in the flow - does it mean that chances of getting the error message like "Transaction is expired" are higher?

after receiving of a blockhash we are preparing other data (for instance, prepareWrappedAccount below)